### PR TITLE
asterisk: Sipsak check fix

### PR DESCRIPTION
**Pull request by x0r3n**
added ocf variable between quotation mark because there are a special character ":"

### CODE REVIEW

**Comment by x0r3n** [comment]
There was an error "ERROR: /usr/lib/ocf/resource.d/heartbeat/asterisk: 1: /usr/lib/ocf/resource.d/heartbeat/asterisk: (type:: not found" but fixed and tested

**Comment by oalbrigt** [comment]
You need to run e.g. `git rebase upstream/master` followed by `git push -f` to get the right base and get rid of the commits listed here that are already merged in upstream.

If you're having issues doing so I can cherry-pick your commit and make a PR for you.

**Comment by x0r3n** [comment]
I tried now rebase, but got "current branch master is up to date". I will be grateful if you can help me with this.)

**Comment by x0r3n** [comment]
> You need to run e.g. `git rebase upstream/master` followed by `git push -f` to get the right base and get rid of the commits listed here that are already merged in upstream.
> 
> If you're having issues doing so I can cherry-pick your commit and make a PR for you.

I tried now rebase, but got "current branch master is up to date". I will be grateful if you can help me with this.)

**Comment by oalbrigt** [comment]
Maybe your remote for upstream is named something? Try running `git remote -v` and look for the name the ones containing "ClusterLabs/resource-agents.git".

**Comment by x0r3n** [comment]
Ok , I think now is ok. Can you check please?

**Comment by oalbrigt** [comment]
Now you got it to a state where I can squash out all the additional commits.

You should be able to get it down to only list your commit (and not the Merge commits and commits you did previously):
https://stackoverflow.com/questions/33502205/how-to-delete-merge-commits

Also we prefer commits to contain e.g. "asterisk: " in the commit message, so we can easily distinguish which commits are related to specific agents.

I've not accepted this for CI yet, so you can try your luck with the suggestoins in the above post without triggering new test jobs.

**Comment by x0r3n** [comment]
> Now you got it to a state where I can squash out all the additional commits.
> 
> You should be able to get it down to only list your commit (and not the Merge commits and commits you did previously): https://stackoverflow.com/questions/33502205/how-to-delete-merge-commits
> 
> Also we prefer commits to contain e.g. "asterisk: " in the commit message, so we can easily distinguish which commits are related to specific agents.
> 
> I've not accepted this for CI yet, so you can try your luck with the suggestoins in the above post without triggering new test jobs.

ok, I tried the above description, hope it looks good now.

**Comment by oalbrigt** [comment]
Now you can do `git rebase -i HEAD~3` and change pick to d or drop for the 2 commits that doesnt contain the quote fix, and it should be fine.

**Comment by x0r3n** [comment]
> Now you can do `git rebase -i HEAD~3` and change pick to d or drop for the 2 commits that doesnt contain the quote fix, and it should be fine.

I tried and if I drop the 2 commit, after it when I wanna push it, I got a 
` ! [rejected]          master -> master (non-fast-forward)
error: failed to push some refs to 'github.com:x0r3n/resource-agents.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.`
and I tried to git pull , then he wanna merge it and put a merge commit message

**Comment by oalbrigt** [comment]
Try `git push -f` to force it. This is due to removing or changing things in your fork instead of only adding like we usually do.

**Comment by x0r3n** [comment]
> Try `git push -f` to force it. This is due to removing or changing things in your fork instead of only adding like we usually do.

ok, it seems good now

**Comment by oalbrigt** [comment]
Perfect. Can you do you `git commit --amend` and add "asterisk: " prefix to the commit message. That way I can merge it without squashing.

In the future it's good to do `git checkout -b asterisk-fix` or similar names and then just git merge or rebase the master and use it for base for the branches.

**Comment by x0r3n** [comment]
> Perfect. Can you do you `git commit --amend` and add "asterisk: " prefix to the commit message. And then push -f again. That way I can merge it without squashing.
> 
> In the future it's good to do `git checkout -b asterisk-fix` or similar names and then just git merge or rebase the master and use it for base for the branches.

Thank you for help :)

**Comment by oalbrigt** [comment]
You're welcome.

**Comment by oalbrigt** [comment]
ok to test

File: heartbeat/asterisk [hunk 1]
```diff
@@ -308,7 +308,7 @@ asterisk_monitor() {
     #   start.
    
     #To avoid the case where the sipsak check runs before the sip starts at the start action
-    SIPCHECK=`sipsak -s $OCF_RESKEY_monitor_sipuri`
+    SIPCHECK=`sipsak -s '$OCF_RESKEY_monitor_sipuri'`
```
Comment (oalbrigt) [inline heartbeat/asterisk:311]: I just noticed you should use double-quotes and not backwards quotes around the command here. \` quotes makes SIPCHECK contain the output of the command, which ocf_run will then try to run later.

Comment (x0r3n) [reply heartbeat/asterisk:311]: True, just tested it, the correct way will to put it to double-quotes (") and delete a ' from inside from variable
 SIPCHECK="sipsak -s $OCF_RESKEY_monitor_sipuri"

Comment (oalbrigt) [reply heartbeat/asterisk]: Looks good now.

Can you squash the commits? Use `git rebase -i HEAD~2` and replace "pick" with "s" on the last line.

Comment (x0r3n) [reply heartbeat/asterisk]: > 

sure.
when I wanna rebase, there are 2 commit, 1st my sip uri fix the second one which I fetched from upstream db2...
Should I drop the second one and just leave with squash my one ?

Comment (oalbrigt) [reply heartbeat/asterisk]: Just squash the 2, and then remove the 2nd line from the new commit message.

Comment (x0r3n) [reply heartbeat/asterisk]: > 

ok, like this. can you check please?

Comment (oalbrigt) [reply heartbeat/asterisk]: You also got the db2 patch that's already been merged in there.

If you do `git checkout HEAD^ heartbeat/db2` and then run `git commit --amend` you can then run `git rebase upstream/master` to get that patch added before your commit.

Comment (x0r3n) [reply heartbeat/asterisk]: > 

hmm, its should look like this ? I did a screen shot about git log [https://imgur.com/a/3IticRY](url)

Comment (oalbrigt) [reply heartbeat/asterisk]: It seems like it's missing the merge commit: commit af7a311021116b8b6d317af4c795745d6c115e38 (HEAD -> master, upstream/master)

**Comment by oalbrigt** [comment]
I squashed the commits for you. Thanks.

**Pull request merged**